test/build-combination/HeaderSelfContain/main.c

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
Could this be handled in the cmake file instead? Seems like you could use the -include argument to gcc to achieve this.

Reply:
That's a good idea to achieve this.
Updated PR by:
1. Add `-include ${header}.h` as compile option.
2. Remove redundant main.c and stubs.c and reuse main.c in Common directory.

Thanks Paul.